File: exercises/beecrowd/1094/1094.go
// https://judge.beecrowd.com/en/problems/view/1094
package main

import (
	"fmt"
)

func main() {
	var testCases int
	fmt.Scan(&testCases)

	var total, rabbits, rats, frogs int

	for i := 0; i < testCases; i++ {
		var quantity int
		var animalType string
		fmt.Scan(&quantity, &animalType)

		total += quantity

		switch animalType {
		case "C":
			rabbits += quantity
		case "R":
			rats += quantity
		case "S":
			frogs += quantity
		}
	}

	fmt.Printf("Total: %d cobaias\n", total)
	fmt.Printf("Total de coelhos: %d\n", rabbits)
	fmt.Printf("Total de ratos: %d\n", rats)
	fmt.Printf("Total de sapos: %d\n", frogs)
	fmt.Printf("Percentual de coelhos: %.2f %%\n", float64(rabbits)*100/float64(total))
	fmt.Printf("Percentual de ratos: %.2f %%\n", float64(rats)*100/float64(total))
	fmt.Printf("Percentual de sapos: %.2f %%\n", float64(frogs)*100/float64(total))
}

File: exercises/beecrowd/1094/1094.ts
// https://judge.beecrowd.com/en/problems/view/1094
import * as fs from "fs";

function main(): void {
	const input = fs.readFileSync('/dev/stdin', 'utf8');
	const lines = input.trim().split(/\r?\n/);

	const testCases = parseInt(lines[0]);
	let total = 0;
	let rabbits = 0;
	let rats = 0;
	let frogs = 0;

	for (let i = 1; i <= testCases; i++) {
		const [qtyStr, type] = lines[i].split(' ');
		const quantity = parseInt(qtyStr);
		total += quantity;

		switch (type) {
			case 'C':
				rabbits += quantity;
				break;
			case 'R':
				rats += quantity;
				break;
			case 'S':
				frogs += quantity;
				break;
		}
	}

	console.log(`Total: ${total} cobaias`);
	console.log(`Total de coelhos: ${rabbits}`);
	console.log(`Total de ratos: ${rats}`);
	console.log(`Total de sapos: ${frogs}`);
	console.log(`Percentual de coelhos: ${(rabbits * 100 / total).toFixed(2)} %`);
	console.log(`Percentual de ratos: ${(rats * 100 / total).toFixed(2)} %`);
	console.log(`Percentual de sapos: ${(frogs * 100 / total).toFixed(2)} %`);
}

main();

